src/ActionMonitor/Monitors/UserMonitor.php

```diff
@@ -135,10 +165,31 @@ public function callback_delete_user( $user_id, $reassign_id ) {
 			return;
 		}
 
+		if ( ! $this->is_published_author( $user_id ) ) {
+			return;
+		}
+
 		// Get the user the posts should be re-assigned to
 		$reassign_user = ! empty( $reassign_id ) ? get_user_by( 'id', $reassign_id ) : null;
 
-		$this->users_before_delete[ $user_id ] = [
+		if ( ! empty( $reassign_user ) ) {
+
+			// @todo: We should get rid of this as it can get expensive to log these actions.
+			// Gatsby Source WordPress should have support for bulk-actions so we can log a single action
+			// such as "DELETE_AUTHOR_AND_REASSIGN_POSTS" and pass the old author ID and new author ID and
+			// Gatsby could do it without an action per modified post.
+			global $wpdb;
+			$post_types = $this->action_monitor->get_tracked_post_types();
+			$post_types = implode( "', '", $post_types );
+			$post_ids = $wpdb->get_col( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_author = %d AND post_type IN ('$post_types')", $user_id ) );
```

Comment:
This needs to be published posts only. Right now it's also creating actions to reassign drafts and trashed posts.

Reply:
Fixed. And included a test to ensure this behavior continues.